projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.spec.ts

```diff
@@ -853,6 +861,101 @@ describe('PreviewsComponent', () => {
         checkPreview(previews[i], previewImages[i], i === 0, DEFAULT_PREVIEW_SIZE);
       }
     });
+
+    [4, 5].forEach((previewNumber: number) => {
+      it(`should display a constant number of previews (${previewNumber}), independent of current image index`, waitForAsync(() => {
+        const configService = fixture.debugElement.injector.get(ConfigService);
+        const previewConfig = Object.assign({}, PREVIEWS_CONFIG_VISIBLE, { number: previewNumber }) as PreviewConfig;
+        configService.setConfig(GALLERY_ID, {
+          previewConfig,
+          accessibilityConfig: KS_DEFAULT_ACCESSIBILITY_CONFIG,
+          slideConfig: SLIDE_CONFIG
+        });
+        comp.id = GALLERY_ID;
+        comp.currentImage = IMAGES[0];
+        comp.images = IMAGES;
+        comp.ngOnInit();
+        expect(comp.previews.length).toBe(previewNumber);
+
+        // click on the second picture
+        comp.currentImage = IMAGES[1]; // set component input
+        comp.ngOnChanges({ // trigger changes manually (not done automatically in tests)
+          currentImage: {
+            previousValue: IMAGES[0],
+            currentValue: IMAGES[1],
+            firstChange: false,
+            isFirstChange: () => false
+          }
+        } as SimpleChanges);
+        // at the time of writing this test, a change is detected within 'images', yet with the same value
+        comp.ngOnChanges({
+          images: {
+            previousValue: IMAGES,
+            currentValue: IMAGES,
+            firstChange: false,
+            isFirstChange: () => false
+          }
+        } as SimpleChanges);
+        expect(comp.previews.length).toBe(previewNumber);
+
+        // click on the third picture
+        comp.currentImage = IMAGES[2]; // set component input
+        comp.ngOnChanges({ // trigger changes manually (not done automatically in tests)
+          currentImage: {
+            previousValue: IMAGES[1],
+            currentValue: IMAGES[2],
+            firstChange: false,
+            isFirstChange: () => false
+          }
+        } as SimpleChanges);
+        // at the time of writing this test, a change is detected within 'images', yet with the same value
+        comp.ngOnChanges({
+          images: {
+            previousValue: IMAGES,
+            currentValue: IMAGES,
+            firstChange: false,
+            isFirstChange: () => false
+          }
+        } as SimpleChanges);
+        expect(comp.previews.length).toBe(previewNumber);
+      }));
+    });
+
+    it('should allow to navigate previews from first to last and back', () => {
```

Comment:
Addresses: when opening the modal and navigating to the last preview by clicking on the right preview arrow, it's impossible to then click on the left preview arrow: isPreventSliding() prevents it